### ISSUE TITLE
A Survey of Methods for Addressing Class Imbalance in Deep-Learning Based Natural Language Processing

### ISSUE THREAD

**Issue by RandyGen** [RandyGen/reading_papers#10]
# A Survey of Methods for Addressing Class Imbalance in Deep-Learning Based Natural Language Processing
https://arxiv.org/pdf/2210.04675.pdf

## Abst
- 多くのNLPタスクはアンバランス
    - 頻度の低いクラスに対する性能がまだ低い
    - 特定のタスクとアンバランスに適したアプローチを見つけることは困難
- DLベースのNLPにおけるクラスの不均衡に関する概要
    - 様々なタイプのクラスインバランスと実世界のクラスインバランスの議論
    - 様々なアプローチ
        - クラスが不均衡なNLPタスクのために明示的に提案されたもの
        - CVやそれらのタスクで評価されたもの
    - 手法の整理
        - サンプリング
        - データ拡張
        - 損失関数の選択
        - 段階的学習
        - モデル設計
    - 未解決の問題について
        - マルチラベルシナリオの処理など
## Intro
- NLPのタスクでは対象のカテゴリ分布がほぼ常に偏っているためクラスのアンバランスが大きな問題
    - 少数クラスのパフォーマンス低下
    - どのカテゴリが重要かはタスクに大きく依存
        - さらに下流の用途に依存する場合も
- 不均衡なデータ設定における性能向上の開発は数十年にわたり活発
    - 最近ではニューラルアプローチの成熟の文脈で勢い増し
- 多種多様なNLPタスクにおいて、トランスフォーマーモデルは、従来のモデルと比較して強いパフォーマンス
    - BERTの登場はNLPにおけるクラスインバランス問題を未解決
- クラスインバランス下でモデル性能を向上させる手法はNLPと関連性があるにもかかわらず、関連する調査はCVのみ

### Losses for Multi-Label Scenarios
- 各ラベルの割り当ては2値決定とみなせる
    - 2値クロスエントロピー（BCE）が希に使用
- 不均衡なマルチラベル分類では2つの問題が発生
    - クラス固有の重みが少数クラスに対する効果がシングルラベルより不明瞭
        - クラスに対して高い重みを使用
            - 少数クラスの重要性が増加し、モデルが少数クラスを予測不可へ
        - BCEにおいてクラス重みをより効果的に活用するためには、
            - ポジティブインスタンスの損失にのみ適用
            - 全てのクラスの正のインスタンスを一様にアップウェイト
            - 与えられたミニバッチに対して、このミニバッチ内の全てのインスタンスに対する損失が全てのクラスに対して全く同じ値を有するようなFL
    - 分類器がすでに正しく負のクラスを予測している場合，それぞれのラベルの対数を減らすことによって損失減少
        - CEは正例のlogitsが負例のlogitsよりはるかに大きくなれば影響小
        - BCEではlogitsが独立して扱われるためこの問題はより深刻
            - 少数派がほぼ負クラスとみなされるため少数派の予測確率低下
- Wuら（2020）はFLのマルチラベル版に基づいて物体検出のための分布バランス損失（DB）を提案
    - 正のクラスと負のクラスの対数を異なる形で変換することにより、負のクラスに対する損失にネガティブトレラント正則化を追加
    - 各logitsが閾値を下回ると負クラスの損失関数が急激に低下
    - インスタンス固有のクラス重みを導入
        - 重みはクラスの頻度、インスタンスの正ラベルの量と頻度を反映
        - Huangら（2021）は2つのロングテールなマルチラベル英語テキスト分類データセットにおいて均一サンプリングを用いた場合でも効果的

### 
## Insights and Future Directions
- ほとんどの研究は単一ラベルのテキスト分類に焦点
    - これらのアプローチを複数ラベルの設定に移行することはしばしば困難
- クラスインバランスはシーケンスラベリングや構文解析などのタスクでも問題
- ベースラインを同じタイプの手法に限定しないこと
    - クラス不均衡設定のための新しいデータ増強手法は、他のデータ増強手法と比較するだけでなく、クラス不均衡のための損失関数とも比較
    - 多くのNLP研究は集約された統計値を報告するだけで、少数クラス対多数クラスの改善への影響を判断するのは困難
        - クラス頻度別に性能を報告するのが一般的
- NLP究者は常にクラスごとの統計量を報告すべき
    - クラスの不均衡に関する研究課題に答えるためのデータを再利用可能
    - 精度の向上のみに焦点を当てるのではなく、関連する少数クラスの性能を分析する必要性
